src/Raven.Server/Documents/Subscriptions/AbstractSubscriptionStorage.cs

```diff
@@ -91,10 +91,10 @@ public SubscriptionState GetSubscriptionById(ClusterOperationContext context, lo
         return GetSubscriptionByName(context, name);
     }
 
-    public static void EnsureValidArchivedBehaviorInSubscriptionState(DatabaseRecord record, ref SubscriptionState subscriptionState)
+    public static ArchivedDataProcessingBehavior HandleNullableArchivedDataProcessingBehavior(DatabaseRecord record, ArchivedDataProcessingBehavior? processingBehavior)
```

Comment:
is this a good place for such a method?
it solves many problems, but I kind of doubt if it's in a good place now

Reply:
this static, essentially, reduces the code by far, solving the same problem in many different places

Reply:
got rid of the static, used the abstract method GetDefaultArchivedDataProcessingBehavior();